app/atomic_svc.py

```diff
@@ -0,0 +1,240 @@
+import json
+import glob
+import hashlib
+import os
+import re
+import shutil
+import yaml
+
+from collections import defaultdict
+from subprocess import DEVNULL, STDOUT, check_call
+
+from app.utility.base_world import BaseWorld
+from app.utility.base_service import BaseService
+
+PLATFORMS = dict(windows='windows', macos='darwin', linux='linux')
+EXECUTORS = dict(command_prompt='cmd', sh='sh', powershell='psh', bash='sh')
+RE_VARIABLE = re.compile('(#{(.*?)})', re.DOTALL)
+
+
+class AtomicService(BaseService):
+    def __init__(self, services):
+        self.log = self.add_service('atomic_svc', self)
+
+        # Atomic Red Team attacks don't come with the corresponding tactic (phase name)
+        # so we need to create a match between techniques and tactics.
+        # This variable is filled by self._populate_dict_techniques_tactics()
+        self.technique_to_tactics = None
```

Comment:
we should initialize as defaultdict(list) instead of None, as that's what we initialize it as when we start (line 105). that would mean you can drop line 105 altogether.

Reply:
fixed in 5b66cb5, I was not sure `defaultdict(list)` would be correctly handled as an empty dict (in `if defaultdict(list)`), but it actually is :)